test/components/KonnectorTile.spec.js

```diff
@@ -0,0 +1,115 @@
+  error,
+  inMaintenance,
+  userError,
+  konnector,
+  route: `/${konnector.slug}`
+})
+
+describe('KonnectorTile component', () => {
+  it('should render correctly if success', () => {
+    const mockProps = getMockProps()
+    const component = shallow(<KonnectorTile {...mockProps} />).getElement()
+    expect(component).toMatchSnapshot()
+  })
+
+  it('should display correct status if update', () => {
+    const mockProps = getMockProps(
+      null,
+      null,
+      Object.assign({}, mockKonnector, { available_version: '5.0.0' })
+    )
+
+    const component = shallow(<KonnectorTile {...mockProps} />).getElement()
+    expect(component).toMatchSnapshot()
+  })
+
+  it('should display correct status if update even if user error', () => {
+    const mockProps = getMockProps(
+      null,
+      new Error('Expected test user error'),
+      Object.assign({}, mockKonnector, { available_version: '5.0.0' })
+    )
+
+    const component = shallow(<KonnectorTile {...mockProps} />).getElement()
+    expect(component).toMatchSnapshot()
+  })
+
+  it('should display correct status if update even if other error', () => {
+    const mockProps = getMockProps(
+      new Error('Expected test error'),
+      null,
+      Object.assign({}, mockKonnector, { available_version: '5.0.0' })
+    )
+
+    const component = shallow(<KonnectorTile {...mockProps} />).getElement()
+    expect(component).toMatchSnapshot()
+  })
+
+  it('should display correct status if update even if in maintenance', () => {
+    const mockProps = getMockProps(
+      null,
+      null,
+      Object.assign({}, mockKonnector, { available_version: '5.0.0' }),
+      true
+    )
+
+    const component = shallow(<KonnectorTile {...mockProps} />).getElement()
+    expect(component).toMatchSnapshot()
+  })
+
+  it('should display correct status if in maintenance but no update', () => {
+    const mockProps = getMockProps(null, null, mockKonnector, true)
+
+    const component = shallow(<KonnectorTile {...mockProps} />).getElement()
+    expect(component).toMatchSnapshot()
+  })
+
+  it('should display correct error status if user error but no update and not in maintenance', () => {
+    const mockProps = getMockProps(null, new Error('Expected test user error'))
+
+    const component = shallow(<KonnectorTile {...mockProps} />).getElement()
+    expect(component).toMatchSnapshot()
+  })
+
+  it('should display correct error status if other error but no update and not in maintenance', () => {
+    const mockProps = getMockProps(new Error('Expected test user error'))
+
+    const component = shallow(<KonnectorTile {...mockProps} />).getElement()
+    expect(component).toMatchSnapshot()
+  })
+
+  it('should display correct error status if no accounts and no errors', () => {
+    const mockProps = getMockProps()
+    mockProps.accountsCount = 0
+
+    const component = shallow(<KonnectorTile {...mockProps} />).getElement()
+    expect(component).toMatchSnapshot()
+  })
```

Comment:
The order of `getMockProps` arguments is difficult to remember.

What I like to do for test cases where there are a lot of combinations : have a setup function creating  the tested element depending on suite global variables.

```js
describe('test', () => {
   let component, error, userError, inMaintenance, konnector

   beforeEach(() => {
     error = null
     userError = null
     inMaintenance = false
     konnector = mockKonnector
   })

   const setup = () => {
     component = shallow(<KonnectorTile {...getMockProps}).getElement()
   }

   it('should display an error', () => {
      error = new Error()
      setup()
      expect(component).toMatchSnapshot()
   })

   it('should display in maintenance, () => {
     inMaintenance = true
      setup()
      expect(component).toMatchSnapshot()
   })
})
```

I personally find it more readable. WDYT ?

Reply:
I agree with you about the `null` usage here, it could be better.
But I am not very comfortable about using global variables across unit tests here, in my my those unit tests must be the must independent as possible and could be run in parallel without any interferences.
For now, let's move and we'll see how to improve that later.